docs(compiler-hooks): adds examples, formatting and missing descriptions (#2840)
webpack/webpack.js.org: src/content/api/compiler-hooks.md

`SyncBailHook`

> Called after the [`entry` configuration](https://webpack.js.org/configuration/entry-context/#entry) from webpack options has been processed.

Reviewer: ```suggestion
Called after the [`entry` configuration](/configuration/entry-context/#entry) from webpack options has been processed.
```

> - Callback Parameters: [`context`](https://webpack.js.org/configuration/entry-context#context), [`entry`](https://webpack.js.org/configuration/entry-context#entry)

Reviewer: ```suggestion
- Callback Parameters: [`context`](/configuration/entry-context/#context), [`entry`](/configuration/entry-context/#entry)
```

```js
compiler.hooks.entryOption.tap('MyPlugin', (context, entry) => {
  /* ... */
});
```

Parameters: `context`, `entry`

### `afterPlugins`

`SyncHook`

Called after setting up initial set of internal plugins.

- Callback Parameters: `compiler`

### `afterResolvers`

`SyncHook`

Triggered after resolver setup is complete.

- Callback Parameters: `compiler`

### `environment`

`SyncHook`

Called while preparing the compiler environment, right after inizializing the plugins in the configuration file.

### `afterEnvironment`

`SyncHook`

Called right after the `environment` hook, when the compiler environment setup is complete.

### `beforeRun`

`AsyncSeriesHook`

Adds a hook right before running the compiler.

- Callback Parameters: `compiler`

### `run`

`AsyncSeriesHook`

> Hook into the compiler before it begins reading ["records"](https://webpack.js.org/configuration/other-options/#recordspath). 

Reviewer: ```suggestion
Hook into the compiler before it begins reading [`records`](/configuration/other-options/#recordspath). 
```

Runs a plugin after a compilation has been created.

- Callback Parameters: `compilation`, `compilationParams`

### `make`

`AsyncParallelHook`

Executed before finishing the compilation.

- Callback Parameters: `compilation`

### `afterCompile`

`AsyncSeriesHook`

Called after finishing and sealing the compilation.  

- Callback Parameters: `compilation`

### `shouldEmit`

`SyncBailHook`

> Called before emitting assets, should return a boolean to say wheter ot not emit.

Reviewer: ```suggestion
Called before emitting assets. Should return a boolean telling whether to emit.
```
